api/api_ethereum.go

```diff
@@ -460,8 +460,13 @@ func (api *EthereumAPI) GetHeaderByHash(ctx context.Context, hash common.Hash) m
 // * When fullTx is true all transactions in the block are returned, otherwise
 //   only the transaction hash is returned.
 func (api *EthereumAPI) GetBlockByNumber(ctx context.Context, number rpc.BlockNumber, fullTx bool) (map[string]interface{}, error) {
+	// Klaytn backend returns error when there is no matched block but
+	// Ethereum returns it as nil without error, so we should return is as nil when there is no matched block.
 	klaytnBlock, err := api.publicBlockChainAPI.b.BlockByNumber(ctx, number)
-	if klaytnBlock != nil && err == nil {
+	if err != nil {
```

Comment:
We don't need this `if` branch because it will end with `nil` return at the bottom of the method.

Reply:
Removed if branch in [this commit](https://github.com/klaytn/klaytn/pull/1261/commits/5c6c7c5ce59b488fc18714acbdf9592cd3ba73ae) PTAL.